src/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.cpp

```diff
@@ -97,22 +97,65 @@ Real Segment::Solution::computeDeltaV(const Real& aSpecificImpulse) const
 
 Mass Segment::Solution::computeDeltaMass() const
 {
-    if (segmentType != Segment::Type::Maneuver)
+    if (this->segmentType != Segment::Type::Maneuver)
     {
         return Mass::Kilograms(0.0);
     }
 
     return Mass::Kilograms(getInitialMass().inKilograms() - getFinalMass().inKilograms());
 }
 
+Array<State> Segment::Solution::reComputeStatesAt(
```

Comment:
Alright, I guess the python naming is aligned with this, fair. I'd call it recomputeStatesAt here nah? That's nit, disregard if this is wrong english